stdlib/public/runtime/Casting.cpp

```diff
@@ -2876,8 +2876,50 @@ id _bridgeAnythingNonVerbatimToObjectiveC(OpaqueValue *src,
 extern "C" const _ObjectiveCBridgeableWitnessTable BRIDGING_CONFORMANCE_SYM;
 #endif
 
+struct ObjCBridgeConformanceMemo {
+  // TODO: If asserts are not being compiled, don't allocate this extra var
+  const Metadata *destType;
+  const _ObjectiveCBridgeableWitnessTable *destBridgeWitness;
+  swift_once_t fetchWitnessOnce;
+
+  const _ObjectiveCBridgeableWitnessTable *
+  findBridgeWitness(const Metadata *T) {
+    struct SetupData {
+      const Metadata *destType;
+      struct ObjCBridgeConformanceMemo *memo;
+    } setupData { T, this };
+
+    swift_once(&fetchWitnessOnce,
+               [](void *data) {
+                 struct SetupData *setupData = (struct SetupData *)data;
```

Comment:
Since we are in c++, can you make this a reinterpret_cast? Also. Why not use auto * on LHS side.